configure

```diff
@@ -513,9 +513,14 @@ else
   CC_TMP="$CC"
   test -n "$OPT_CC" && OTHER_CC= || OTHER_CC="gcc cc"
   for CC in "$CC_TMP" $OTHER_CC; do
+    cc_name_tmp=
     if run_cmd "$CC -v"; then
       cc_name_tmp=$($CC -v 2>&1 | tail -n 1 | cut -d ' ' -f 1)
-      if test "$cc_name_tmp" = "gcc"; then
+    elif run_cmd "$CC --version"; then
+      cc_name_tmp=$($CC --version 2>&1 | head -n 1 | cut -d ' ' -f 1)
+    fi
+    if test "${cc_name_tmp}"; then
```

Comment:
POSIX [recommends](https://pubs.opengroup.org/onlinepubs/9699919799/utilities/test.html) to use `test -n "$1"` instead of `test "$1"`